src/zen/websocket_server.cpp

```diff
@@ -954,6 +1043,21 @@ static int getblock(const CBlockIndex *pindex, std::string& strHex)
 }
 
 
+static int getheader(const CBlockIndex *pindex, std::string& strHex)
+{
+    CDataStream ss(SER_NETWORK, PROTOCOL_VERSION);
+    {
+        LOCK(cs_main);
+        CBlock block;
+        if (!ReadBlockFromDisk(block, pindex)) {
+            LogPrint("ws", "%s():%d - error: could not read block from disk\n", __func__, __LINE__);
+            return WsHandler::READ_ERROR;
+        }
+        ss << block.GetBlockHeader();
```

Comment:
The analogous GetBlockHeader() could be called also on pindex ptr directly, without reading the block from disk